kubernetes/istio_details_service.go

```diff
@@ -16,34 +16,33 @@ import (
 // If serviceName param is provided, it filters all the Istio objects pointing to a particular service.
 // It returns an error on any problem.
 func (in *IstioClient) GetIstioDetails(namespace string, serviceName string) (*IstioDetails, error) {
-	var virtualServices, destinationRules []IstioObject
-	var virtualServicesErr, destinationRulesErr error
-	var wg sync.WaitGroup
-	wg.Add(2)
 
-	go func() {
-		defer wg.Done()
-		virtualServices, virtualServicesErr = in.GetVirtualServices(namespace, serviceName)
-	}()
+	wg := sync.WaitGroup{}
+	errChan := make(chan error, 4)
```

Comment:
Mmm probably here you can use same pattern of errChan buffer size 1 and use a select in the check for errors ?
But I like as it is, as it's more readable.

Reply:
I could, but the idea to keep it with size 4 is that there's no way to generate more errors as we don't fetch some unknown number of items, such as namespaces.